R/aaa-auto.R

```diff
@@ -826,7 +826,7 @@ get_widest_path_impl <- function(graph, from, to, weights=NULL, mode=c("out", "i
   res
 }
 
-get_widest_paths_impl <- function(graph, from, to=V(graph), weights=NULL, mode=c("out", "in", "all", "total")) {
+get_widest_paths_impl <- function(graph, from, to=V(graph), weights, mode=c("out", "in", "all", "total")) {
```

Comment:
The `weights=NULL` seems useful, let's keep it.

Reply:
I'm going to have to check out today, but in brief:

 - This function _requires_ weights, passing `NULL` in C is not valid.
 - Because there's a high risk that people will pass `NULL`, the C function does check for it, and reports a nice error (instead of crashing).
 - In R, the default _should_ be `NULL` (R's `NULL` this time) because this signifies that weights should be taken from edge attributes. It doesn't mean "use no weights". That's `NA`.
 - This function is _not_ exposed in R at the moment.
 - When it gets exposed, the quick solution to deal with this is what you are doing here: override the interface definition in R and add that `=NULL`. This shouldn't be done in C, only in R. But a nicer solution would be to patch Stimulus to treat `weights` specially, for multiple reasons, including this one. Then it would know that R needs the default of `NULL` even when this is not an optional parameter in C.